swing/src/net/sf/openrocket/gui/dialogs/ComponentAnalysisDialog.java

```diff
@@ -534,65 +531,78 @@ public void stateChanged(EventObject e) {
 			}
 		}
 
+//		!? - needs to be reimplemented:
 		Map<RocketComponent, AerodynamicForces> aeroData =
 				aerodynamicCalculator.getForceAnalysis(configuration, conditions, set);
-		Map<RocketComponent, Coordinate> massData =
-				massCalculator.getCGAnalysis(configuration);
-
+		Map<String, CMAnalysisEntry> cmMap= massCalculator.getCMAnalysis(configuration);
 
-		cgData.clear();
+		stabData.clear();
 		dragData.clear();
 		rollData.clear();
-		for (RocketComponent c : configuration.getActiveComponents()) {
-			if ( c instanceof AxialStage ) {
+
+		for(final RocketComponent comp: configuration.getAllComponents()) {
+			// // this is actually redundant, because the analysis will not contain inactive stages.
+			// if (!configuration.isComponentActive(comp)) {
+			// 	continue;
+			// }
+
+			CMAnalysisEntry cmEntry = cmMap.get(comp.getName());
+			if (null == cmEntry) {
+				System.err.println("Could not find entry for component: " + comp.getName());
```

Comment:
Send to logs instead of stderr

Reply:
Temporary debugging line.  Will be removed before merging.

Reply:
(meaning, I don't believe this is significant enough to warrant a log line.)

Reply:
Removed.